Jenkinsfile

```diff
@@ -536,4 +510,193 @@ def upDir(path) {
+$sw = [Diagnostics.Stopwatch]::StartNew()
+try {
+    Push-Location "build/$env:BUILD_DIR"
+    if ($env:NINJA_BUILD -eq "true") {
+        cmake -G"Ninja" -Dtarget="$env:COMPILER.$env:TARGET" -DCMAKE_VERBOSE_MAKEFILE=ON ../..
+    }
+    else {
+        cmake -G"Visual Studio 15 2017 Win64" -Dtarget="$env:COMPILER.$env:TARGET" -DCMAKE_VERBOSE_MAKEFILE=ON ../..
+    }
+    if ($LastExitCode -ne 0) { throw "CMake failed" }
+
+    ## as of 01/2018, DO NOT USE cmake to run the actual build step. for some
+    ## reason, cmake spawning the build under jenkins causes MSBUILD/ninja to
+    ## get stuck at the end of the build. Perhaps cmake is spawning
+    ## incorrectly or failing to pass certain params
+
+    if ($env:NINJA_BUILD -eq "true") {
+        ninja -j $env:NUMBER_OF_PROCESSORS -v
+    }
+    else {
+        msbuild /fl /m /nr:false /p:Configuration="$env:CONFIG_TYPE" /p:Platform=x64 /p:GenerateFullPaths=True /v:normal /nologo /clp:"ShowCommandLine;DisableConsoleColor" "$env:PROJECT_NAME.vcxproj"
+    }
+    if ($LastExitCode -ne 0) { throw "CMake build failed" }
+
+    $exe = "./$env:CONFIG_TYPE/$env:PROJECT_NAME"
+    if ($env:NINJA_BUILD -eq "true") {
+        $exe = "./$env:PROJECT_NAME"
+    }
+    "Exe is at $exe"
+    $params = '--unittest', '--quiet', '--unittest-log'
+    if ($env:PARALLEL_TESTS -eq "true") {
+        $params = $params += "--unittest-jobs=$env:NUMBER_OF_PROCESSORS"
+    }
+    & $exe $params
+    if ($LastExitCode -ne 0) { throw "Unit tests failed" }
+}
+catch {
+    throw
+}
+finally {
+    $sw.Stop()
+    $sw.Elapsed
+    Pop-Location
+}
+'''
+}
+
+// the shell command used for building an RPM
+def rpmBuildCmd(bldlabel) {
+'''\
+#!/bin/bash
+set -ex
+log_file=''' + "${bldlabel}.txt" + '''
+exec 3>&1 1>>${log_file} 2>&1
+
+# Vault Steps
+SECRET_ID=$(cat /.vault/rippled-build-role/secret-id)
+export VAULT_TOKEN=$(/usr/local/ripple/ops-toolbox/vault/vault_approle_auth -r ${ROLE_ID} -s ${SECRET_ID} -t)
+/usr/local/ripple/ops-toolbox/vault/vault_get_sts_token.py -r rippled-build-role
+
+head $SIGNER_PUBLIC_KEYS
+
+mkdir -p rpm-out
+
+docker pull "${docker_image}"
+
+echo "Running build container"
+
+docker run --rm \
+-v $PWD/rpm-out:/opt/rippled-rpm/out \
+-v $SIGNER_PUBLIC_KEYS:/opt/rippled-rpm/public-keys.txt \
+-e "GIT_COMMIT=$git_commit" \
+-e "GIT_REMOTE=$git_remote" \
+-e "RPM_RELEASE=$rpm_release" \
+"${docker_image}"
+
+. rpm-out/build_vars
+
+cd rpm-out
+tar xvf rippled-*.tar.gz
+ls -la *.rpm
+#################################
+## for now we don't want the src
+## and debugsource rpms for testing
+## or archiving...
+#################################
+rm rippled-debugsource*.rpm
+rm *.src.rpm
+mkdir rpm-main
+cp *.rpm rpm-main
+cd rpm-main
+cd ../..
+
+cat > test_rpm.sh << "EOL"
+#!/bin/bash
+
+function error {
+  echo $1
+  exit 1
+}
+
+yum install -y yum-utils
+rpm -i /opt/rippled-rpm/*.rpm
+rc=$?; if [[ $rc != 0 ]]; then
+  error "error installing rpms"
+fi
+
+/opt/ripple/bin/rippled --unittest
```

Comment:
You could temporarily change this to something like `--unittest=ValidatorKeys` or rebase on Scott's fix to verify that we make it through the rpm build when tests pass.